### PR TITLE
[TIMOB-19946] Implement Ti.UI.AlertDialog.hide() functionality

### PR DESCRIPTION
**Pull request by garymathews**
- Implemented the ability to hide active alert dialog's

###### TEST CASE
```Javascript
var a = Ti.UI.createAlertDialog({title: 'Alert', message: 'This will hide in 3 seconds'});
a.show();

// hide alert after 3 seconds
setTimeout(function () {
    a.hide();
}, 3000);

// show alert after another 3 seconds
setTimeout(function () {
    a.show();
}, 6000);
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-19946)

### CODE REVIEW

**Comment by garymathews** [comment]
test this please

**Comment by sgtcoolguy** [comment]
@garymathews This doesn't appear to work for me. I tested on Windows 10 mobile and got an "Access is denied" exception when the second show() call happened. This is because the hide() call cancelled the task and removed the dialog from the queue, but *the dialog never closed!* It appears to be a limitation/difference between Store/desktop and Phone/mobile? http://stackoverflow.com/questions/28795850/how-to-dismiss-a-messagedialog-from-code-in-windows-phone-8-1

So either we should:
 - use macros to turn off the hide() impl used here for phones to avoid the scenario I'm seeing where it dumps the dialog from the queue without closing it. and maybe log something saying you can't hide alert dialogs programatically on phone;
 - **or** we may need to look into using a different mechanism for the dialog other than MessageDialog. I saw remarks pointing to https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.controls.contentdialog.aspx if you need to programatically close the dialog.

**Comment by garymathews** [comment]
test this please

File: Source/UI/src/AlertDialog.cpp [hunk 1]
```diff
@@ -105,5 +106,19 @@ namespace TitaniumWindows
 				detail::ThrowRuntimeError("Ti.UI.AlertDialog", "Exception during show()");
 			}
 		}
+
+		void AlertDialog::hide() TITANIUM_NOEXCEPT
+		{
+#ifndef IS_WINDOWS_PHONE
```
Comment (sgtcoolguy) [inline Source/UI/src/AlertDialog.cpp]: I still think this won't guard against windows 10 mobile...

**Comment by garymathews** [comment]
test this please

File: Source/UI/src/AlertDialog.cpp [hunk 2]
```diff
@@ -105,5 +106,19 @@ namespace TitaniumWindows
 				detail::ThrowRuntimeError("Ti.UI.AlertDialog", "Exception during show()");
 			}
 		}
+
+		void AlertDialog::hide() TITANIUM_NOEXCEPT
+		{
+#if defined(IS_WINDOWS_PHONE)
```
Comment (garymathews) [inline Source/UI/src/AlertDialog.cpp]: It seems the ```IS_WINDOWS_MOBILE``` condition is still satisfied on a Windows 10 Universal Desktop application. I've removed it for now so ```IS_WINDOWS_PHONE``` will have to surpass.

**Comment by infosia** [comment]
Tested this on Windows 10 Mobile but I got ""Access is denied" exception.

**Comment by garymathews** [comment]
@infosia Updated.

**Comment by garymathews** [comment]
test this please

**Comment by infosia** [comment]
Looks good to me. Will merge once all Jenkins tests go through.

**Comment by infosia** [comment]
test this please

**Pull request merged**